tc/sgx/trusted_worker_manager/enclave/signup_enclave.cpp

```diff
@@ -247,27 +399,35 @@ tcf_err_t ecall_UnsealEnclaveData(
             enclaveData->get_public_data().c_str(),
             enclaveData->get_public_data_size());
     } catch (tcf::error::Error& e) {
-        SAFE_LOG(TCF_LOG_ERROR, "Error in Avalon enclave(ecall_UnsealEnclaveData): %04X -- %s",
+        SAFE_LOG(TCF_LOG_ERROR,
+            "Error in Avalon enclave(ecall_UnsealEnclaveData): %04X -- %s",
             e.error_code(), e.what());
         ocall_SetErrorMessage(e.what());
         result = e.error_code();
     } catch (...) {
-        SAFE_LOG(TCF_LOG_ERROR, "Unknown error in Avalon enclave(ecall_UnsealEnclaveData)");
+        SAFE_LOG(TCF_LOG_ERROR,
+            "Unknown error in Avalon enclave(ecall_UnsealEnclaveData)");
         result = TCF_ERR_UNKNOWN;
     }
 
     return result;
 }  // ecall_UnsealEnclaveData
 
+
 // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 // XX Internal helper functions                                      XX
 // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 
 // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
-void CreateSignupReportData(const char* pOriginatorPublicKeyHash,
-    EnclaveData* enclaveData,
-    sgx_report_data_t* pReportData) {
-    // We will put the following in the report data SHA256(PPK|PEK|OPK_HASH)
+void CreateSignupReportData(uint64_t worker_mode, const uint8_t* ext_data,
+    EnclaveData* enclave_data, sgx_report_data_t* report_data) {
+
+    // We will put the following in the report data
+    // SINGLETON_ENCLAVE - SHA256(PUB SIG KEY)
+    // KME_ENCLAVE - SHA256(PUB SIG KEY|EXT_DATA) where EXT_DATA contains
+    //               MRENCLAVE value of associated WPE
+    // WPE_ENCLAVE - SHA256(PUB ENC KEY|EXT DATA) where EXT_DATA contains
+    //               verification key generated by KME
```

Comment:
Above comments about REPORTDATA is not correct. Please clearly mention what goes in REPORTDATA[0-31] and REPORTDATA[32-63]

Reply:
Ok, will do the correction